sqlx-postgres/src/types/mod.rs

```diff
@@ -179,6 +179,14 @@
 //! enum Mood { Sad = 0, Ok = 1, Happy = 2 }
 //! ```
 //!
+//! Rust enumerations may also be defined to be represented as a string using `type_name = "text"`.
+//! The following type expects a SQL type of `TEXT` and will convert to/from the Rust enumeration.
+//!
```

Comment:
This maybe should mention that an error can occur if attempting to decode a value not contained within the enum definition. You never know what people might assume it does otherwise.

Reply:
Yeah. This seems to apply to all the cases listed here. S,o I add a note at the end.

From my experience, it's still possible that users don't find the doc at the end. But duplicating it to each case seems awkward. If you'd like to mention it at the beginning of this section, I can make the change.

Reply:
Yeah, it's just more likely when decoding an arbitrary `TEXT` value. It's a little more obvious when there's a mismatch between a SQL enum and a Rust enum. Regardless, I agree that it technically applies to all of them.